### PR TITLE
Add orchestration test suite (118 tests) and CI/CD pipeline

### PR DESCRIPTION
**Pull request by pranavpai**
## Summary

- **6 new test modules** covering every public function in the orchestration layer — the heart of the 7-phase pipeline
- **GitHub Actions CI workflow** running ruff lint + unittest on Python 3.11/3.12 for every push and PR
- **Fix `.gitignore` bug** where `tests/` was accidentally ignored, blocking new test files from being tracked
- **Add ruff per-file-ignores** for E402/I001 in tests to match the existing `sys.path` import convention

### Test Coverage Added

| Module | What it tests | Tests |
|--------|--------------|-------|
| `test_contracts.py` | Envelope & ArtifactMeta validation, serialization, roundtrips | 22 |
| `test_state_machine.py` | PipelineStateStore, OrchestratorStateMachine validation, write perms | 18 |
| `test_router.py` | SkillRouter keywords (English + Chinese), confidence levels, fallback | 24 |
| `test_write_guard.py` | Phase-scoped filesystem isolation, guard installation, async execute | 14 |
| `test_model_profiles.py` | ModelProfileResolver, fallback chains, MetricsLogger | 12 |
| `test_phase_completion_extended.py` | output_exists edge cases, markers, all 6 failure patterns | 17 |

### What this does NOT touch

- Zero changes to runtime code (`0xclaw/runtime/`)
- Zero changes to existing test files
- Zero new dependencies
- All tests follow existing unittest + TemporaryDirectory conventions

## Test plan

- [x] All 126 tests pass locally (`python -m unittest discover -s tests -p "test_*.py" -v`)
- [x] Ruff lint clean on all new files (`ruff check tests/`)
- [x] Pre-existing `test_main_handoff.py` error is unchanged (missing `loguru` in system Python — unrelated)
- [x] CI workflow runs on this PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by Rqcker — COMMENTED** [review]
Looks like added Github auto action for Python unit tests. 

This is indeed a good habit, but it will increase the time of each commit. The team and community will discuss and decide whether to merge.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: pip install -e ".[dev]"
+
+      - name: Lint
+        run: ruff check .
```
[the remark this answers is not in the document]
Comment (Rqcker) [reply .github/workflows/ci.yml]: GitHub Actions in this PR only need to care about the /tests and unit test folder. So Copilot may be right? We don't need to loop the whole project? @pranavpai

**Comment by pranavpai** [comment]
Good catch @Rqcker — you're absolutely right. `ruff check .` would fail on pre-existing lint issues in the main codebase (E402/I001 in `0xclaw/main.py` etc.) that are outside the scope of this PR.

I've scoped the lint step to `ruff check tests` so CI only validates what this PR introduces. Fix pushed in the latest commit.

**Review by Rqcker — APPROVED** [review]
Let's try via github auto action

**Comment by pranavpai** [comment]
Thanks for the merge @Rqcker! If the CI overhead feels too heavy on pushes, happy to scope it to PRs only — just let me know. Looking forward to contributing more.

**Comment by Rqcker** [comment]
> Thanks for the merge @Rqcker! If the CI overhead feels too heavy on pushes, happy to scope it to PRs only — just let me know. Looking forward to contributing more.

No worries, but action looks like error raised cuz model version mismatch, but it's fine we just change the model to GLM-4.5 today. I can fix it:

```
Traceback (most recent call last):
  File "/home/runner/work/0xClaw/0xClaw/tests/test_model_profiles.py", line 35, in test_resolve_returns_correct_coding_profile
    self.assertEqual(profile.model, "glm-4.7")
AssertionError: 'glm-4.5' != 'glm-4.7'
- glm-4.5
?       ^
+ glm-4.7
?       ^
```
